src/Build/TelemetryInfra/TelemetryDataUtils.cs

```diff
@@ -0,0 +1,319 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+namespace Microsoft.Build.Framework.Telemetry
+{
+    internal static class TelemetryDataUtils
+    {
+        public static IActivityTelemetryDataHolder? AsActivityDataHolder(this IWorkerNodeTelemetryData? telemetryData, bool includeTasksDetails, bool includeTargetDetails)
+        {
+            if (telemetryData == null)
+            {
+                return null;
+            }
+
+            List<TelemetryItem> telemetryItems = new(4);
+
+            if (includeTasksDetails)
+            {
+                telemetryItems.Add(new TelemetryItem("Tasks",
+                    JsonSerializer.Serialize(telemetryData.TasksExecutionData, _serializerOptions), false));
+            }
+
+            if (includeTargetDetails)
+            {
+                telemetryItems.Add(new TelemetryItem("Targets",
+                    JsonSerializer.Serialize(telemetryData.TargetsExecutionData, _serializerOptions), false));
+            }
+
+            TargetsSummary targetsSummary = new();
+            targetsSummary.Initialize(telemetryData.TargetsExecutionData);
+            telemetryItems.Add(new TelemetryItem("TargetsSummary",
+                JsonSerializer.Serialize(targetsSummary, _serializerOptions), false));
+
+            TasksSummary tasksSummary = new();
+            tasksSummary.Initialize(telemetryData.TasksExecutionData);
+            telemetryItems.Add(new TelemetryItem("TasksSummary",
+                JsonSerializer.Serialize(tasksSummary, _serializerOptions), false));
+
+            return new NodeTelemetry(telemetryItems);
+        }
+
+        private static JsonSerializerOptions _serializerOptions = CreateSerializerOptions();
+
+        private static JsonSerializerOptions CreateSerializerOptions()
+        {
+            var opt = new JsonSerializerOptions
+            {
+                WriteIndented = true,
```

Comment:
This should probably be removed. (was there for easier review of data during testing)